### PR TITLE
noexecstack addition to CMakeLists

### PR DESCRIPTION
**Pull request by tcormackMW**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -363,6 +363,17 @@ if (APPLE AND US_COMPILER_APPLE_CLANG)
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
 endif()
 
+#-----------------------------------------------------------------------------
+# Set stack as non executable
+#-----------------------------------------------------------------------------
+
+# g3232705: Add flag to force non executable stack on gcc builds
```
Comment (jeffdiclemente) [inline CMakeLists.txt:370]: fix comment, `g3232705` has no meaning here.

**Review by jeffdiclemente — CHANGES_REQUESTED** [review]
looks like there are macos build failures to investigate as well.

File: CMakeLists.txt (hunk 1 above)
Comment (tcormackMW) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -363,6 +363,17 @@ if (APPLE AND US_COMPILER_APPLE_CLANG)
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
 endif()
 
+#-----------------------------------------------------------------------------
+# Set stack as non executable
+#-----------------------------------------------------------------------------
+
+# Add flag to force non executable stack on gcc builds
+# The -Wl,-z noexecstack link option on all gcc builds will
```
Comment (jeffdiclemente) [inline CMakeLists.txt:371]: you can build using clang on linux. does this linker option work for clang as well as gcc?

**Review by jeffdiclemente — APPROVED** [review]

**Pull request merged**